### DISCUSSION TITLE
Can I modify the m-code for a given dataset from C# script?

### DISCUSSION THREAD

**Discussion by dpesante** [TabularEditor/TabularEditor3#856] [🙋♀️TE3 Q&A] [answered]
We are looking to automate some of our m-code from C# script. Is this even doable?

**Comment by mlonsk** [comment, marked as answer]
Hi @dpesante 
It is possible. You can access the partition's expression and for example replace it with another expression. 

An example of accessing the partition for the Customers table is has follows: 
`Model.Tables["Customers"].Partitions["Customers"].Expression`

You can also try to record a Macro which I sometimes use to understand how to access these properties
![image](https://user-images.githubusercontent.com/11939114/226606962-42974996-3875-4755-b23d-9d0ff7908173.png)
